terraform/modules/matrix-service/infra/converter_batch.tf

```diff
@@ -0,0 +1,221 @@
+  service_role = "${aws_iam_role.AWSBatchServiceRole.arn}"
+  compute_resources {
+    type = "SPOT"
+    bid_percentage = 100
+    spot_iam_fleet_role = "${aws_iam_role.AmazonEC2SpotFleetRole.arn}"
+    max_vcpus = 256
+    min_vcpus = 0
+    // You must set desired_vcpus otherwise you get error: "desiredvCpus should be between minvCpus and maxvCpus"
+    // However this is actually not settable in AWS.  It will not let you change it.
+    // Here we use an external data source to dynamically set the desired vcpus to match current state.
+    desired_vcpus = "${data.external.converter_desired_vcpus.result.desired_vcpus}"
+    instance_type = [
+      "m4"
+    ]
+    image_id = "${var.converter_cluster_ami_id}"
+    subnets = ["${data.aws_subnet.default.*.cidr_block}"]
+    security_group_ids = [
+      "${aws_default_vpc.default.default_security_group_id}"
+    ]
+    ec2_key_pair = "matrix-service-${var.deployment_stage}"
+    instance_role = "${aws_iam_instance_profile.ecsInstanceRole.arn}"
+  }
+  depends_on = [
+    "aws_iam_role_policy_attachment.AWSBatchServiceRole"
+  ]
+}
+
+resource "aws_batch_job_queue" "converter_job_queue" {
+  name = "dcp-matrix-converter-queue-${var.deployment_stage}"
+  compute_environments = [
+    "${aws_batch_compute_environment.converter_compute_env.arn}"]
+  priority = 1
+  state = "ENABLED"
+}
+
+resource "aws_iam_policy" "converter_job_policy" {
+  name = "dcp-matrix-converter-job-${var.deployment_stage}"
+  policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Sid": "s3",
+        "Effect": "Allow",
+        "Action": [
+          "s3:*"
+        ],
+        "Resource": [
+          "arn:aws:s3:::dcp-matrix-service-results-${var.deployment_stage}",
+          "arn:aws:s3:::dcp-matrix-service-results-${var.deployment_stage}/*"
+        ]
+      }
+    ]
+}
+POLICY
+}
+
+resource "aws_iam_role" "converter_job_role" {
+  name = "dcp-matrix-converter-job-${var.deployment_stage}"
+  assume_role_policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "",
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "ecs-tasks.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+        },
+        {
+            "Sid": "",
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "lambda.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+POLICY
+}
+
+resource "aws_iam_role_policy_attachment" "converter_job_role" {
+  role = "${aws_iam_role.converter_job_role.name}"
+  policy_arn = "${aws_iam_policy.converter_job_policy.arn}"
+}
+
+resource "aws_batch_job_definition" "converter_job_def" {
+    name = "dcp-matrix-converter-job-definition-${var.deployment_stage}"
+    type = "container"
+    container_properties = <<CONTAINER_PROPERTIES
+{
+    "command": [],
+    "image": "humancellatlas/matrix-converter:latest",
```

Comment:
Maybe better if I tag an image?

Reply:
@mckinsel I'm actually using slightly different code/image to get it to work with batch. That WIP PR is coming up shortly. I'll be deploying the image to humancellatlas/matrix-converter in dockerhub. I'm including a concept of VERSION. I agree. I'll replace latest with version here.